nilearn/regions/rena_clustering.py

```diff
@@ -150,6 +151,129 @@ def _make_edges_and_weights(X, mask_img):
     return edges, weights
 
 
+def _compute_weights_surface(X, mask, edges):
+    """Compute the weights for each edge using Euclidean distance.
+
+    Parameters
+    ----------
+    X : ndarray, shape = [n_samples, n_features]
+        Training data.
+
+    mask : boolean
+        Object used for masking the data.
+
+    edges : ndarray
+
+    Returns
+    -------
+    weights : ndarray
+        Weights corresponding to all edges in the mask.
+        shape: (n_edges,).
+
+    """
+    n_samples, _ = X.shape
+    shape = mask.shape
+
+    data = np.empty((shape[0], n_samples))
+    for sample in range(n_samples):
+        data[:, sample] = unmask_from_to_3d_array(X[sample].copy(), mask)
+
+    data_i = data[edges[0]]
+    data_j = data[edges[1]]
+    weights = np.sum((data_i - data_j) ** 2, axis=-1).ravel()
+
+    return weights
+
+
+def _make_edges_surface(faces, mask):
+    """Create the edges set: Returns a list of edges for a surface mesh.
+
+    Parameters
+    ----------
+    faces : ndarray
+        The vertex indices corresponding the mesh triangles.
+
+    mask : boolean
```

Comment:
I don't see how to make sense of mask in this function, given that the edge ordering is unclear at that point ?

Reply:
maks is on vertices, right ? Then it makes sense.  But the dosctring has to be improved.